src/main/java/com/example/bookstore/service/category/CategoryServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.example.bookstore.service.category;
+
+import com.example.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.bookstore.dto.category.CategoryDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.BookMapper;
+import com.example.bookstore.mapper.CategoryMapper;
+import com.example.bookstore.model.Book;
+import com.example.bookstore.model.Category;
+import com.example.bookstore.repository.category.CategoryRepository;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryMapper categoryMapper;
+    private final CategoryRepository categoryRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable)
+                .stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(()
+                -> new EntityNotFoundException("Can't find category by id: " + id));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryDto categoryDto) {
+        Category category = categoryMapper.toModel(categoryDto);
+        category.setId(id);
+
+        Category categoryFromDb = categoryRepository.findById(category.getId()).orElseThrow(()
+                -> new EntityNotFoundException("Can't find category with id: " + category.getId()));
+        categoryFromDb.setName(category.getName());
+        categoryFromDb.setDescription(category.getDescription());
+        return categoryMapper.toDto(categoryRepository.save(categoryFromDb));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoriesId(Long id) {
+        Category category = categoryRepository.findCategoryWithBooks(id);
```

Comment:
Everything can be fetched in one query. Additionally, this method more suits to the services related to the book, not to the category